validmind/vm_models/dataset/dataset.py

```diff
@@ -537,24 +537,34 @@ def __init__(
             target_class_labels (dict, optional): The class labels for the target columns. Defaults to None.
             date_time_index (bool, optional): Whether to use date-time index. Defaults to False.
         """
+
+        VMInput.__init__(self)
+
+        self.input_id = input_id
+        self._raw_dataset = raw_dataset.values
+
         index = None
         if isinstance(raw_dataset.index, pd.Index):
             index = raw_dataset.index.values
+        self.index = index
 
-        super().__init__(
-            raw_dataset=raw_dataset.values,
-            input_id=input_id,
-            model=model,
-            index_name=raw_dataset.index.name,
-            index=index,
-            columns=raw_dataset.columns.to_list(),
-            target_column=target_column,
-            extra_columns=extra_columns,
-            feature_columns=feature_columns,
-            text_column=text_column,
-            target_class_labels=target_class_labels,
-            date_time_index=date_time_index,
-        )
+        # Store the DataFrame directly
+        self._df = raw_dataset.copy()
```

Comment:
I think we can do the following:

```suggestion
        self._df = raw_dataset
```

Since the `.df` accessor already does this:

```python
    @property
    def df(self) -> pd.DataFrame:
        """
        Returns the dataset as a pandas DataFrame.

        Returns:
            pd.DataFrame: The dataset as a pandas DataFrame.
        """
        # only include feature, text and target columns
        # don't include internal pred and prob columns
        columns = self.feature_columns.copy()

        # text column can also be a feature column so don't add it twice
        if self.text_column and self.text_column not in columns:
            columns.append(self.text_column)

        if self.extra_columns.extras:
            # add user-defined extra columns
            columns.extend(self.extra_columns.extras)

        if self.target_column:
            # shouldn't be a feature column but add this to be safe
            assert self.target_column not in columns
            columns.append(self.target_column)

        # return a copy to prevent accidental modification
        return as_df(self._df[columns]).copy()
```

Reply:
Follow up (from our call):

As another short term fix, I suggest allowing the `init_dataset()` to have an option to not create copies under conditions of limited resources or when dealing with large datasets. The user will have to be aware that they shouldn't be making changes to these dataframes inside tests.